CI test failing on sending coverage report

**What happened?**
Avalon's CI test keeps failing when it need to send the coverage report via `mayapy` which is now an outdated Python..

See https://travis-ci.org/github/getavalon/core/builds/712074966#L548-L572
```log
Ran 32 tests in 11.158s
OK
/usr/autodesk/maya/lib/python2.7/site-packages/urllib3/util/ssl_.py:394: SNIMissingWarning: An HTTPS request has been made, but the SNI (Server Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  SNIMissingWarning,
/usr/autodesk/maya/lib/python2.7/site-packages/urllib3/util/ssl_.py:143: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecurePlatformWarning,
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/autodesk/maya/lib/python2.7/site-packages/coveralls/__init__.py", line 95, in wear
    skip_ssl_verify=args.skip_ssl_verify,
  File "/usr/autodesk/maya/lib/python2.7/site-packages/coveralls/api.py", line 8, in pos
    return requests.post(url, files={'json_file': json_file}, verify=(not skip_ssl_verify))
  File "/usr/autodesk/maya/lib/python2.7/site-packages/requests/api.py", line 119, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/usr/autodesk/maya/lib/python2.7/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/autodesk/maya/lib/python2.7/site-packages/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/autodesk/maya/lib/python2.7/site-packages/requests/sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "/usr/autodesk/maya/lib/python2.7/site-packages/requests/adapters.py", line 514, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='coveralls.io', port=443): Max retries exceeded with url: /api/v1/jobs (Caused by SSLError(SSLError(1, '_ssl.c:507: error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure'),))
The command ". test_docker.sh" exited with 1.
```

### Proposal

Currently the test is invoked in Maya docker container, also generates coverage report and sending it there.
The test should be ran by `mayapy`, but we should define a path for it to save coverage file in somewhere else and pick up by another python process to send the report.

--- a/.coveragerc
+++ b/.coveragerc
@@ -1,6 +1,12 @@
 [run]
 source = avalon
 
+[paths]
+# Remap source root path in docker
+source =
+    avalon
+    /workspace/avalon
+
 [report]
 include = *avalon*
 omit = 

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,6 +3,10 @@ language: python
 python:
   - 3.7
 
+env:
+  global:
+    - COVERALLS_SERVICE_NAME=travis-ci
+
 services:
   - docker
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -21,7 +21,7 @@ script:
 
 after_success:
   - coverage combine
-  - coveralls
+  - coveralls debug
 
 before_deploy:
   - sudo chmod -R 777 docs/build/

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,15 +3,11 @@ language: python
 python:
   - 3.7
 
-env:
-  global:
-    - COVERALLS_SERVICE_NAME=travis-ci
-
 services:
   - docker
 
 install:
-  - pip install coverage coveralls
+  - pip install coverage python-coveralls
   - docker run --name avalon-mongo -d mongo
   - . build_docker.sh
 
@@ -21,7 +17,7 @@ script:
 
 after_success:
   - coverage combine
-  - coveralls debug
+  - coveralls
 
 before_deploy:
   - sudo chmod -R 777 docs/build/

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,18 +1,23 @@
 language: python
 
 python:
-  - 2.7
+  - 3.7
 
 services:
   - docker
 
 install:
+  - pip install coverage coveralls
   - docker run --name avalon-mongo -d mongo
-  - docker build -t avalon/core -f Dockerfile-maya2016 .
+  - docker build -t avalon/core:maya -f Dockerfile-maya2016 .
 
 script:
   - . test_docker.sh
   - . build_docs.sh
+
+after_success:
+  - coverage combine
+  - coveralls
 
 before_deploy:
   - sudo chmod -R 777 docs/build/

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,7 +7,8 @@ services:
   - docker
 
 install:
-  - pip install coverage python-coveralls
+  - pip install coverage==4.5.4
+  - pip install python-coveralls
   - docker run --name avalon-mongo -d mongo
   - . build_docker.sh
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,7 +9,7 @@ services:
 install:
   - pip install coverage coveralls
   - docker run --name avalon-mongo -d mongo
-  - docker build -t avalon/core:maya -f Dockerfile-maya2016 .
+  - . build_docker.sh
 
 script:
   - . test_docker.sh

--- a/Dockerfile-maya2016
+++ b/Dockerfile-maya2016
@@ -13,7 +13,7 @@ RUN wget https://bootstrap.pypa.io/get-pip.py && \
 		six \
 		sphinxcontrib-napoleon
 
-# Avoid creation of auxilliary files
+# Avoid creation of auxiliary files
 ENV PYTHONDONTWRITEBYTECODE=1
 
 WORKDIR /workspace

--- a/Dockerfile-maya2016
+++ b/Dockerfile-maya2016
@@ -22,5 +22,5 @@ ENV PYTHONDONTWRITEBYTECODE=1
 WORKDIR /workspace
 
 ENTRYPOINT \
-	PYTHONPATH=$(pwd):$PYTHONPATH mayapy -u run_maya_tests.py && \
-	mayapy -c "import sys, os, coveralls;coveralls.wear() if os.getenv('TRAVIS_JOB_ID') else sys.stdout.write('Skipping coveralls')"
+	PYTHONPATH=$(pwd):$PYTHONPATH && \
+	mayapy -u run_maya_tests.py

--- a/Dockerfile-maya2016
+++ b/Dockerfile-maya2016
@@ -5,16 +5,13 @@ RUN wget https://bootstrap.pypa.io/get-pip.py && \
 	mayapy -m pip install \
 		nose \
 		nose-exclude \
-		coverage==4.5.4 \
-		pyblish-base==1.4.2 \
+		coverage \
+		pyblish-base \
 		pyblish-maya \
 		pymongo \
-		# Sphinx 1.8.0 fail to build doc, pin version to 1.7.9
-		# see https://github.com/sphinx-doc/sphinx/issues/5417
-		sphinx==1.7.9 \
+		sphinx \
 		six \
-		sphinxcontrib-napoleon \
-		python-coveralls
+		sphinxcontrib-napoleon
 
 # Avoid creation of auxilliary files
 ENV PYTHONDONTWRITEBYTECODE=1

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -54,7 +54,7 @@ jobs:
 
   - job: MacOS
     pool:
-      vmImage: "macOS-10.13"  # High-Sierra
+      vmImage: "macOS-10.15"  # Catalina
     strategy:
       matrix:
         Python27:

--- a/build_docker.sh
+++ b/build_docker.sh
@@ -1,1 +1,1 @@
-docker build -t avalon/core -f Dockerfile-maya2016 .
+docker build -t avalon/core:maya -f Dockerfile-maya2016 .

--- a/build_docs.sh
+++ b/build_docs.sh
@@ -1,11 +1,11 @@
 docker run \
-	--rm \
-	-v $(pwd):/workspace \
-	--link avalon-mongo:mongo \
-	-e AVALON_SILENT \
-	-e AVALON_MONGO=mongodb://mongo:27017 \
-	-e PYTHONPATH=/workspace:/usr/local/lib/python2.6/site-packages \
-	--entrypoint mayapy \
-	--workdir /workspace/docs \
-	avalon/core \
-	-m sphinx source build -E -v
+  --rm \
+  --volume $(pwd):/workspace \
+  --link avalon-mongo:mongo \
+  --env AVALON_SILENT \
+  --env AVALON_MONGO=mongodb://mongo:27017 \
+  --env PYTHONPATH=/workspace:/usr/local/lib/python2.6/site-packages \
+  --entrypoint mayapy \
+  --workdir /workspace/docs \
+  avalon/core:maya \
+  -m sphinx source build -E -v

--- a/test_docker.sh
+++ b/test_docker.sh
@@ -6,6 +6,9 @@ docker run \
   --link avalon-mongo:mongo \
   --env AVALON_SILENT \
   --env AVALON_MONGO=mongodb://mongo:27017 \
+  --env TRAVIS_JOB_ID \
+  --env TRAVIS_BRANCH \
+  --env COVERALLS_REPO_TOKEN \
   avalon/core:maya
 # Rename coverage data file and will be combined later
 mv .coverage .coverage.maya

--- a/test_docker.sh
+++ b/test_docker.sh
@@ -1,9 +1,13 @@
+
+# Run Maya test
 docker run \
-	--rm \
-	-v $(pwd):/workspace \
-	--link avalon-mongo:mongo \
-    -e COVERALLS_REPO_TOKEN \
-    -e TRAVIS_JOB_ID \
-	-e AVALON_SILENT \
-	-e AVALON_MONGO=mongodb://mongo:27017 \
-	avalon/core
+  --rm \
+  --volume $(pwd):/workspace \
+  --link avalon-mongo:mongo \
+  --env AVALON_SILENT \
+  --env AVALON_MONGO=mongodb://mongo:27017 \
+  avalon/core:maya
+# Rename coverage data file and will be combined later
+mv .coverage .coverage.maya
+
+# docker run other DCC tools..

--- a/test_docker.sh
+++ b/test_docker.sh
@@ -6,9 +6,6 @@ docker run \
   --link avalon-mongo:mongo \
   --env AVALON_SILENT \
   --env AVALON_MONGO=mongodb://mongo:27017 \
-  --env TRAVIS_JOB_ID \
-  --env TRAVIS_BRANCH \
-  --env COVERALLS_REPO_TOKEN \
   avalon/core:maya
 # Rename coverage data file and will be combined later
 mv .coverage .coverage.maya

